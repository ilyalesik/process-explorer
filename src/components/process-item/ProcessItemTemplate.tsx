import React from "react";
import "./processItemTemplate.css";

export const ProcessItemTemplate = (props: {
	name: React.Node;
	cpu: React.Node;
	memory: React.Node;
	children?: React.Node;
	close: React.Node;
}) => {
	const { name, cpu, memory, children, close } = props;

	return (
		<>
			<div className="processItemTemplate">
				<div className="processItemTemplate__column processItemTemplate__name">{name}</div>
				<div className="processItemTemplate__column">
					<p class="processItemTemplate__label">cpu</p>
					{cpu}
				</div>
				<div className="processItemTemplate__column">
					<p class="processItemTemplate__label">memory</p>
					{memory}
				</div>
				<div className="processItemTemplate__closeContainer">{close}</div>
			</div>
			<div class="processItemTemplate__children">{children}</div>
		</>
	);
};
