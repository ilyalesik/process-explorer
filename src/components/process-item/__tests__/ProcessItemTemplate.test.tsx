import * as React from "react";
import * as renderer from "react-test-renderer";
import { ProcessItemTemplate } from "../ProcessItemTemplate";

describe("<ProcessItemTemplate />", () => {
	it("ProcessItemTemplate renders correctly", () => {
		const tree = renderer.create(
			<ProcessItemTemplate name={"name"} cpu={"cpu"} memory={"memory"} close={"close"}>
				{"children"}
			</ProcessItemTemplate>
		);
		expect(tree).toMatchSnapshot();
	});
});
