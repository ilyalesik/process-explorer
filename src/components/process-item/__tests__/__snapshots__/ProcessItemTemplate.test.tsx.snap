// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProcessItemTemplate /> ProcessItemTemplate renders correctly 1`] = `
Array [
  <div
    className="container"
  >
    <div
      className="column name"
    >
      name
    </div>
    <div
      className="column"
    >
      <p
        class="label"
      >
        cpu
      </p>
      cpu
    </div>
    <div
      className="column"
    >
      <p
        class="label"
      >
        memory
      </p>
      memory
    </div>
    <div
      className="closeContainer"
    >
      close
    </div>
  </div>,
  <div
    class="children"
  >
    children
  </div>,
]
`;
